odl/space/fspace.py

```diff
@@ -1036,129 +613,389 @@ def _gaussian_fun(x):
+
+                s = 0
+                for ind in range(len(x)):
+                    s = s + (x[ind] - mins[ind]) / (maxs[ind] - mins[ind])
+
+                return s
+
+            yield ('Grad all', self.element(_all_gradient_fun))
+
+    def __repr__(self):
+        """Return ``repr(self)``."""
+        inner_str = '{!r}'.format(self.domain)
+        dtype_str = dtype_repr(self.out_dtype)
+
+        if self.field == RealNumbers():
+            if self.out_dtype == np.dtype('float64'):
+                pass
+            else:
+                inner_str += ', out_dtype={}'.format(dtype_str)
+
+        elif self.field == ComplexNumbers():
+            if self.out_dtype == np.dtype('complex128'):
+                inner_str += ', field={!r}'.format(self.field)
+            else:
+                inner_str += ', out_dtype={}'.format(dtype_str)
+
+        else:  # different field, name explicitly
+            inner_str += ', field={!r}'.format(self.field)
+            inner_str += ', out_dtype={}'.format(dtype_str)
+
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+    def __str__(self):
+        """Return ``str(self)``."""
+        inner_str = '{}'.format(self.domain)
+        dtype_str = dtype_repr(self.out_dtype)
+
+        if self.field == RealNumbers():
+            if self.out_dtype == np.dtype('float64'):
+                pass
+            else:
+                inner_str += ', out_dtype={}'.format(dtype_str)
+
+        elif self.field == ComplexNumbers():
+            if self.out_dtype == np.dtype('complex128'):
+                inner_str += ', field={!r}'.format(self.field)
+            else:
+                inner_str += ', out_dtype={}'.format(dtype_str)
+
+        else:  # different field, name explicitly
+            inner_str += ', field={!r}'.format(self.field)
+            inner_str += ', out_dtype={}'.format(dtype_str)
+
+        return '{}({})'.format(self.__class__.__name__, inner_str)
+
+    @property
+    def element_type(self):
+        """`FunctionSpaceElement`"""
+        return FunctionSpaceElement
+
+
+class FunctionSpaceElement(LinearSpaceElement):
+
+    """Representation of a `FunctionSpace` element."""
+
+    def __init__(self, fspace, fcall):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        fspace : `FunctionSpace`
+            Set of functions this element lives in.
+        fcall : callable
+            The actual instruction for out-of-place evaluation.
+            It must return a `FunctionSpace.range` element or a
+            `numpy.ndarray` of such (vectorized call).
+        """
+        super(FunctionSpaceElement, self).__init__(fspace)
+
+        # Determine which type of implementation fcall is
+        if isinstance(fcall, FunctionSpaceElement):
+            call_has_out, call_out_optional, _ = _dispatch_call_args(
+                bound_call=fcall._call)
+
+        # Numpy Ufuncs and similar objects (e.g. Numba DUfuncs)
```

Comment:
Prefer to move this comment to after the `elif` statement and remove the empty line above